modules/rollup-plugin-workbox-inject/index.js

```diff
@@ -31,39 +33,74 @@ const chalk = require('chalk');
 function workboxInject(config) {
   return {
     name: 'workbox-inject', // this name will show up in warnings and errors
-    async generateBundle(opts, bundle) {
-      const options = validate(config, injectManifestSchema);
-      // Make sure we leave swSrc and swDest out of the precache manifest.
-      for (const file of [options.swSrc, options.swDest]) {
-        options.globIgnores.push(
-          rebasePath({
-            file,
+    async generateBundle(outputOptions, bundle) {
+      // injectionPoint isn't supported by getManifestSchema, but we need it.
+      const injectionPoint = config.injectionPoint || defaults.injectionPoint;
+      delete config.injectionPoint;
+
+      const options = validate(config, getManifestSchema);
+
+      // globDirectory might not be set, as developers can just provide
+      // additionalManifestEntries and forgo globbing.
+      if (options.globDirectory) {
+        const outputDir = outputOptions.dir ||
+          upath.dirname(upath.resolve(outputOptions.file));
+
+        // Make sure we leave the source service worker and any output of this
+        // specific Rollup compilation out of the precache manifest.
+        for (const chunk of Object.values(bundle)) {
+          const fileRebasedToGlobDirectory = rebasePath({
+            file: upath.resolve(outputDir, chunk.fileName),
             baseDirectory: options.globDirectory,
-          }),
-        );
+          });
+          // Sourcemaps might be disabled, but adding something extra to
+          // globIgnores doesn't hurt.
+          const mapFile = `${fileRebasedToGlobDirectory}.map`;
+
+          options.globIgnores.push(
+            fileRebasedToGlobDirectory,
+            mapFile,
+          );
+
+          // If set, this should be the path to the original source module.
+          if (chunk.facadeModuleId) {
+            const moduleRebasedToGlobDirectory = rebasePath({
+              file: chunk.facadeModuleId,
+              baseDirectory: options.globDirectory,
+            });
+            options.globIgnores.push(moduleRebasedToGlobDirectory);
+          }
+        }
       }
-      const globalRegexp = new RegExp(escapeRegexp(options.injectionPoint), 'g');
 
       const { manifestEntries, count, warnings, size } = await getFileManifestEntries(options);
 
       const manifestString = stringify(manifestEntries);
 
-      const bundles = Object.keys(bundle);
+      for (const chunk of Object.values(bundle)) {
+        // This might actually be an asset (which won't have a code property)
+        // and not a chunk. If so, continue.
+        if (!chunk.code) {
+          continue;
+        }
+
+        if (chunk.code.includes(injectionPoint)) {
```

Comment:
I'm just going to check for `includes()` here rather than creating a `RegExp`. `workbox-build` goes the `RegExp` route so that it could detect whether there are multiple injection point matches in the same file and fail if there are, but your plugin isn't doing that.

Reply:
Should we be supporting multiple injection point matches?

Reply:
So, I don't know how strict you want to get here.

I don't think it's very useful to inject a manifest in to the same file more than once, because even if you need to use it in multiple places, you can just do

```
const manifest = self.__WB_MANIFEST;

// Reuse manifest however you want.
```

We go through the trouble of actually enforcing a single injection point in `workbox-build`'s `injectManifest` mode, but the code in this PR just replaces the first instance, and will leave any others as `self.__WB_MANIFEST`.

Reply:
(That reminds me, though, that this plugin should probably throw an `Error` if it doesn't find a match for `injectionPoint` anywhere, so that it's not a silent failure.)